corehq/apps/app_manager/util.py

```diff
@@ -408,8 +408,9 @@ def get_cloudcare_session_data(domain_name, form, couch_user):
     return session_data
 
 
-def update_form_unique_ids(app_source, form_ids_by_xmlns=None):
+def update_form_unique_ids(app_source, ids_map):
     from corehq.apps.app_manager.models import form_id_references, jsonpath_update
+    ids_map = defaultdict(lambda: uuid.uuid4().hex, ids_map)
```

Comment:
I think it was clearer when the assignment to random id was done below, via `form_ids_by_xmlns.get(form['xmlns'], uuid.uuid4().hex)`. The old expression fits better with my verbal description of what's going on here, which is "Look up the form, and if not found, give it a fresh id."

Reply:
True, though that requires making a second map old-to-new (or updating the dict passed in, which would be gross), since the `reference_path` business below needs to know what ID was used here.

I guess it's probably worth the verbosity for the added clarity.